server/service/devices.go

```diff
@@ -335,3 +337,79 @@ func transparencyURL(ctx context.Context, request interface{}, svc fleet.Service
 
 	return transparencyURLResponse{RedirectURL: transparencyURL}, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Get Current Device's MDM Apple Enrollment Profile
+////////////////////////////////////////////////////////////////////////////////
+
+type getDeviceMDMManualEnrollProfileRequest struct {
+	Token string `url:"token"`
+}
+
+func (r *getDeviceMDMManualEnrollProfileRequest) deviceAuthToken() string {
+	return r.Token
+}
+
+type getDeviceMDMManualEnrollProfileResponse struct {
+	// Profile field is used in hijackRender for the response.
+	Profile []byte
+
+	Err error `json:"error,omitempty"`
+}
+
+func (r getDeviceMDMManualEnrollProfileResponse) hijackRender(ctx context.Context, w http.ResponseWriter) {
+	// TODO(mna): does this filename look ok?
+	// make the browser download the content to a file
+	w.Header().Add("Content-Disposition", `attachment; filename="fleet-mdm-enrollment-profile.mobileconfig"`)
```

Comment:
@noahtalerman does this filename (`fleet-mdm-enrollment-profile.mobileconfig`) look good to you? This is the name that the file would have on download of the manual enrollment profile. Note that generally, it should open the profile settings automatically so the user would have minimal interaction with the file itself, but still we need to give it a name.

Reply:
> does this filename (fleet-mdm-enrollment-profile.mobileconfig) look good to you?

@mna yeah this looks good to me!

> generally, it should open the profile settings automatically so the user would have minimal interaction with the file itself

I don't think it's automatic. Unless something changed (which is possible!). 

My understanding is the user will have to manually double click the profile, then manually open System Settings, and then do some other things to install it. The steps that end user has to take are outlined in the Figma page here: https://www.figma.com/file/hdALBDsrti77QuDNSzLdkx/%F0%9F%9A%A7-Fleet-EE-(dev-ready%2C-scratchpad)?node-id=9983%3A318194&t=9SQ4WHP3DkJqGBlw-1

Reply:
Gotcha. I'm not 100% sure since I don't have a mac and can't test it out, but the content type `application/x-apple-aspen-config` (which we set on the response of this request) appears to be understood by Apple and from what I can google about it (it was already used in the prototype code), just clicking to open the downloaded file should automatically open the profile settings application. 

THanks for confirming the file name!

Reply:
> just clicking to open the downloaded file should automatically open the profile settings application.

Oh...interesting. Last time I tried this out it didn't automatically open settings. Maybe there was a bug in macOS when I tried. 

That said, we'll be tested the "turn on MDM" flow extensively so there will be opportunity to confirm if it works.

It would be fantastic if it's fixed.